packages/cdktf/lib/iterator.ts

```diff
@@ -0,0 +1,200 @@
+import { TerraformStack } from ".";
+import { ComplexList, ComplexMap } from "./complex-computed-list";
+import { Fn } from "./terraform-functions";
+import { propertyAccess, ref } from "./tfExpression";
+import { Lazy, Token } from "./tokens";
+
+export interface IIterator {
+  /**
+   * @internal used by TerraformResource to set the for_each expression
+   */
+  _getForEachExpression(): any;
+}
+
+export abstract class Iterator implements IIterator {
+  /**
+   * @internal used by TerraformResource to set the for_each expression
+   */
+  abstract _getForEachExpression(): any;
+
+  /**
+   * Creates a new iterator from a list
+   */
+  public static fromList(
+    list: string[] | number[] | ComplexList
```

Comment:
This doesn't cover all types of lists.

Reply:
Can you help me finding out which ones are missing?

Reply:
A list of booleans is represented as `Array<boolean | IResolvable> | IResolvable`.
There's also `MapList`.

I think maps have some missing ones as well. For example, `String/Number/Boolean/Any Map`.

Reply:
added ✅ Thank you for helping!